components/siteleaf/sources/new-collection/new-collection.mjs

```diff
@@ -0,0 +1,50 @@
+import base from "../common/base.mjs";
+import app from "../../siteleaf.app.mjs";
+
+export default {
+  ...base,
+  key: "siteleaf-new-collection",
+  type: "source",
+  name: "New Collection",
+  description: "Emit new event when a new collection is created",
+  version: "0.0.1",
+  props: {
+    ...base.props,
+    siteId: {
+      propDefinition: [
+        app,
+        "siteId",
+      ],
+    },
+  },
+  methods: {
+    ...base.methods,
+    async fetchEvents() {
```

Comment:
The `fetchEvents()` method is nearly identical across the sources. I think we can reuse most of this code into a common method in the `base` file for the sources.

An individual `getData` or similar method to fetch `listCollections`, `listPages`, etc. and `getSummary` method would take care of the parts that differ between the sources.